canvas_modules/common-canvas/__tests__/common-canvas/common-canvas-test.js

```diff
@@ -31,54 +33,82 @@ describe("CommonCanvas renders correctly", () => {
 		canvasController = new CanvasController();
 	});
 
+	it("should render one <CanvasBottomPanel/> component when showBottomPanel is true", () => {
+		const config = {};
+		const wrapper = createCommonCanvas(config, canvasController, { showBottomPanel: true }, {});
+		expect(wrapper.find(CanvasBottomPanel)).to.have.length(1);
+		expect(canvasController.isBottomPanelOpen() === true).to.be.true;
+	});
+
+	it("should render one <CanvasBottomPanel/> component when showBottomPanel is false", () => {
+		const config = {};
+		const wrapper = createCommonCanvas(config, canvasController, { showBottomPanel: false }, {});
+		expect(wrapper.find(CanvasBottomPanel)).to.have.length(1);
+		expect(canvasController.isBottomPanelOpen() === false).to.be.true;
+	});
+
+	it("should render one <CommonCanvasRightFlyout/> component when showRightFlyout is true", () => {
+		const config = {};
+		const wrapper = createCommonCanvas(config, canvasController, {}, { showRightFlyout: false });
+		expect(wrapper.find(CommonCanvasRightFlyout)).to.have.length(1);
+		expect(canvasController.isRightFlyoutOpen() === true).to.be.false;
+	});
+
+	it("should render one <CommonCanvasRightFlyout/> component when showRightFlyout is false", () => {
+		const config = {};
+		const wrapper = createCommonCanvas(config, canvasController, {}, { showRightFlyout: true });
+		expect(wrapper.find(CommonCanvasRightFlyout)).to.have.length(1);
+		expect(canvasController.isRightFlyoutOpen() === false).to.be.false;
+	});
+
 	it("should render one <CanvasContents/> component", () => {
 		const config = {};
-		const wrapper = createCommonCanvas(config, canvasController);
+		const wrapper = createCommonCanvas(config, canvasController, {}, {});
```

Comment:
The `createCommonCanvas` function should be able to handle the `canvasParams` parameter when it is undefined so you only need to pass an object when there is a field set within it.

Reply:
Understood !! I tried to pass them but it keeps failing... however I replicated the other instances' behavior like config.. i declared them as empty objects and pass them accordingly, and it seems to pass